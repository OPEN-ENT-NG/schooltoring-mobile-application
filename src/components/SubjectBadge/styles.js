import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  item: {
    padding: 5,
    margin: 5,
    alignSelf: "flex-start"
  },
  itemText: {
    color: COLORS.WHITE
  }
});
