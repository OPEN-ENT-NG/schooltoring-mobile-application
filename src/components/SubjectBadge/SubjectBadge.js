import React, { Component } from "react";
import {
  View,
  Text} from "react-native";
import PropTypes from "prop-types";

import Touchable from "../Touchable/Touchable";

import styles from "./styles";

export default class SubjectBadge extends Component {
  render() {
    return (
      <Touchable
        onPress={() =>
          this.props.hasOwnProperty("onPress") && this.props.onPress()
        }
      >
        <View style={[styles.item, this.props.style]}>
          <Text style={styles.itemText}>{this.props.title}</Text>
        </View>
      </Touchable>
    );
  }
}

SubjectBadge.propTypes = {
  onPress: PropTypes.func,
  style: PropTypes.object,
  title: PropTypes.string.isRequired
};
