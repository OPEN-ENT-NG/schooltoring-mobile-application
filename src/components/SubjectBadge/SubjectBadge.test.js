import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";

import SubjectBadge from "./SubjectBadge";

describe("Render", () => {
  test("SubjectBadge renders without crashing", () => {
    const rendered = shallow(<SubjectBadge title="Test" />);
    expect(rendered).toMatchSnapshot();
  });
  test("Error logged if no title given", () => {
    const PropTypes = jest.spyOn(global.console, "error");
    const rendered = shallow(<SubjectBadge />);
    expect(rendered).toMatchSnapshot();
    expect(PropTypes).toHaveBeenCalled();
  });

  test("Rendered title is the same than the title prop given", () => {
    const title = "Badge title";
    const rendered = shallow(<SubjectBadge title={title} />);

    expect(
      rendered
        .find(Text)
        .first()
        .contains(title)
    );
  });
});
