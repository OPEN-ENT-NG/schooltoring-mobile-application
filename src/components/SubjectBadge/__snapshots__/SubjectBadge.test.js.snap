// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Error logged if no title given 1`] = `
<Touchable
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "margin": 5,
          "padding": 5,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
        }
      }
    />
  </View>
</Touchable>
`;

exports[`Render SubjectBadge renders without crashing 1`] = `
<Touchable
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "margin": 5,
          "padding": 5,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      Test
    </Text>
  </View>
</Touchable>
`;
